src/napi/js_native_ext_api_v8.cpp

```diff
@@ -0,0 +1,273 @@
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+#include "env-inl.h"
+
+#include "V8JsiRuntime_impl.h"
+#include "js_native_api_v8.h"
+#include "js_native_ext_api.h"
+#include "public/ScriptStore.h"
+
+static struct napi_env_scope__ {
+  napi_env_scope__(v8::Isolate *isolate, v8::Local<v8::Context> context)
+      : isolate_scope(isolate ? new v8::Isolate::Scope(isolate) : nullptr),
+        context_scope(
+            !context.IsEmpty() ? new v8::Context::Scope(context) : nullptr) {}
+
+  napi_env_scope__(napi_env_scope__ const &) = delete;
+  napi_env_scope__ &operator=(napi_env_scope__ const &) = delete;
+
+  napi_env_scope__(napi_env_scope__ &&other)
+      : isolate_scope(other.isolate_scope), context_scope(other.context_scope) {
+    other.isolate_scope = nullptr;
+    other.context_scope = nullptr;
+  }
+
+  napi_env_scope__ &operator=(napi_env_scope__ &&other) {
+    if (this != &other) {
+      napi_env_scope__ temp{std::move(*this)};
+      Swap(other);
+    }
+    return *this;
+  }
+
+  void Swap(napi_env_scope__ &other) {
+    std::swap(isolate_scope, other.isolate_scope);
+    std::swap(context_scope, other.context_scope);
+  }
+
+  ~napi_env_scope__() {
+    if (context_scope) {
+      delete context_scope;
+    }
+
+    if (isolate_scope) {
+      delete isolate_scope;
+    }
+  }
+
+ private:
+  v8::Isolate::Scope *isolate_scope{nullptr};
+  v8::Context::Scope *context_scope{nullptr};
+};
+
+napi_status napi_ext_create_env(
+    napi_ext_env_attributes attributes,
+    napi_env *env) {
+  v8runtime::V8RuntimeArgs args;
+  args.trackGCObjectStats = false;
+  args.enableTracing = false;
+  args.enableJitTracing = false;
+  args.enableMessageTracing = false;
+  args.enableLog = false;
+  args.enableGCTracing = false;
+
+  if ((attributes & napi_ext_env_attribute_enable_gc_api) != 0) {
+    args.enableGCApi = true;
+  }
+
+  if ((attributes & napi_ext_env_attribute_ignore_unhandled_promises) != 0) {
+    args.ignoreUnhandledPromises = true;
+  }
+
+  auto runtime = std::make_unique<v8runtime::V8Runtime>(std::move(args));
+
+  auto context = v8impl::PersistentToLocal::Strong(runtime->GetContext());
+  *env = new napi_env__(context);
+
+  // Let the runtime exists. It can be accessed from the Context.
+  runtime.release();
```

Comment:
>runtime [](start = 2, length = 7)

Do we plan to share a runtime between React native and node modules or any other consumers of napi ?